packages/devextreme/testing/tests/DevExpress.ui.widgets/speechToText.tests.js

```diff
@@ -0,0 +1,619 @@
+        });
+
+        assert.ok(!this.$element.hasClass(SPEECH_TO_TEXT_LISTENING_CLASS), 'missing enabled handled');
+    });
+});
+
+QUnit.module('Options', moduleConfig, () => {
+    QUnit.test('should update button icon and text based on state', function(assert) {
+        this.reinit({
+            startText: 'Start Recording',
+            stopText: 'Stop Recording',
+            startIcon: 'user',
+            stopIcon: 'check',
+        });
+
+        const $button = this.getButton();
+        const $buttonText = $button.find(`.${BUTTON_TEXT_CLASS}`);
+
+        let $icon = $button.find(`.${ICON_CLASS}`);
+
+        assert.strictEqual($buttonText.text(), 'Start Recording', 'initial text displayed');
+        assert.ok($icon.hasClass(`${ICON_CLASS}-user`), 'initial icon displayed');
+
+        $button.trigger('dxclick');
+
+        assert.strictEqual($buttonText.text(), 'Stop Recording', 'text updated after click');
+
+        $icon = $button.find(`.${ICON_CLASS}`);
+
+        assert.ok($icon.hasClass(`${ICON_CLASS}-check`), 'icon updated after click');
+    });
+
+    QUnit.test('should support button styling options', function(assert) {
+        this.reinit({
+            stylingMode: 'contained',
+            type: 'danger',
+        });
+
+        const $button = this.getButton();
+
+        assert.ok($button.hasClass(BUTTON_MODE_CONTAINED_CLASS), 'styling mode applied');
+        assert.ok($button.hasClass(BUTTON_DANGER_CLASS), 'button type applied');
+    });
+
+    QUnit.test('should handle icon options change at runtime', function(assert) {
+        const buttonInstance = this.getButtonInstance();
+
+        this.instance.option('startIcon', 'home');
+        assert.strictEqual(buttonInstance.option('icon'), 'home', 'start icon updated');
+
+        this.getButton().trigger('dxclick');
+
+        this.instance.option('stopIcon', 'gear');
+        assert.strictEqual(buttonInstance.option('icon'), 'gear', 'stop icon updated');
+    });
+
+    QUnit.test('should handle text options change at runtime', function(assert) {
+        const buttonInstance = this.getButtonInstance();
+
+        this.instance.option('startText', 'Begin');
+        assert.strictEqual(buttonInstance.option('text'), 'Begin', 'start text updated');
+
+        this.getButton().trigger('dxclick');
+
+        this.instance.option('stopText', 'End');
+        assert.strictEqual(buttonInstance.option('text'), 'End', 'stop text updated');
+    });
+
+    QUnit.test('should handle width and height options', function(assert) {
+        this.reinit({
+            width: 200,
+            height: 50,
+        });
+
+        const buttonInstance = this.getButtonInstance();
+
+        assert.strictEqual(buttonInstance.option('width'), 200, 'width applied to button');
+        assert.strictEqual(buttonInstance.option('height'), 50, 'height applied to button');
+    });
+
+    QUnit.test('should handle stylingMode changes at runtime', function(assert) {
```

Comment:
Why do you test `stylingMode` but not test `type` for example? Or width and height but not focusStateEnabled?
Maybe it's better to write a single test like "button-related option should be passed to button at runtime change"?